### PR TITLE
Updated docs

### PR DESCRIPTION
**Pull request by johandahlberg**
Mostly updated documentation, and some very minor API changes made to make things a bit clearer.

### CODE REVIEW

File: checkQC/handlers/qc_handler.py [hunk 1]
```diff
@@ -1,81 +1,166 @@
...
         self.message = msg
         self.ordering = ordering
         self.data = data
 
-    def __str__(self):
-        return "Fatal QC error: {}".format(self.message)
+    def type(self):
+        """
+        Should be implemented by the subclass.
+        :return: String with the type of the report, e.g. "error" or "waring"
```
Comment (matrulda) [inline checkQC/handlers/qc_handler.py]: typo: "waring" --> "warning"

File: checkQC/handlers/qc_handler.py [hunk 2]
```diff
@@ -1,81 +1,166 @@
...
-        return self.__str__()
+    def type(self):
+        return "warning"
 
-    def as_dict(self):
-        return {'type': 'warning', 'message': self.message, 'data': self.data}
+
+class Subscriber(object):
+    """
+    Subscriber defined the behaviour necessary to subscribe to data from a Parser. The implementing subclass
```
Comment (matrulda) [inline checkQC/handlers/qc_handler.py]: "defines"?

File: checkQC/handlers/qc_handler.py [hunk 3]
```diff
@@ -1,81 +1,166 @@
...
+        :param qc_config: dict containing the configuration for the QCHandler. Should have keys 'error' and 'warning'
+        """
+        super().__init__()
         self._exit_status = 0
         self.qc_config = qc_config
 
     def custom_configuration_validation(self):
         """
         Override this method in subclass to provide additional configuration behaviour.
         Raise a `ConfigurationError` if there is a problem with the provided config.
```
Comment (matrulda) [inline checkQC/handlers/qc_handler.py]: This line (152) might be unnecessary?

File: checkQC/handlers/qc_handler.py [hunk 4]
```diff
@@ -1,81 +1,166 @@
...
+        :raises: ConfigurationError if there is a problem with the configuration
         """
         pass
 
     def validate_configuration(self):
+        """
+        This method will validate the configuration which has been passed to the QCHandler. This should be called
+        by the class making us of this class. It will not be called automatically e.g. at object creation.
```
Comment (matrulda) [inline checkQC/handlers/qc_handler.py]: Typo: "us" --> "use"

Comment (matrulda) [reply checkQC/handlers/qc_handler.py]: Is it possible to change the first "class" to "sub class" or similar to avoid repeating the same word?

**Review by matrulda — APPROVED** [review]
Great! See minor comments, all related to the docs.

**Comment by johandahlberg** [comment]
I've fixed the changes you proposed @MatildaAslin. Feel free to merge this. There will be more PRs like this until, which needs to be written to make sure that the API docs are complete.

**Comment by matrulda** [comment]
:+1:

**Pull request merged**